### DISCUSSION TITLE
Question - not bug.  Using switcher with multiple sources and multiple audio

### DISCUSSION THREAD

**Discussion by TheSashmo** [CESNET/UltraGrid#180] [Q&A]
I have been experimenting with the switcher option, which works great by the way.  But it can't switch audio.  I was testing with one sdi input and one testcard with testcard sound, but I can either get my testcard audio or sdi audio, but I can't switch video with audio.  

I know thats not really the way it should work, since in a production setup, you would be using mixed audio into the source, but I figure that I would ask anyways.

**Comment by MartinPulec** [comment]
Well, this is a bit tricky - AFAIK the audio part here was intended that N-th device is selected with eg. `-s embedded:<N>` which will then capture. That looks inconvenient for the switcher but there are more aggregating capturers, eg. _swmix_ for which that works perfectly.

Not sure if it helps in your setup (I'd need at least the command-line options) but, I've pushed a change that allow the _switcher_ to switch also audio, provided that it is captured from vidcap and of _"the same kind"_, eg. _embedded_, usage eg.:

`uv -t switcher -s embedded -t decklink:device=0 -t aja:device=0`



Martin

**Comment by TheSashmo** [comment]
Thanks Martin I will check this out.
